ttyplot.c

```diff
@@ -591,10 +591,10 @@ int main(int argc, char *argv[]) {
         };
         const int select_ret = select(select_nfds, &read_fds, NULL, NULL, &timeout);
 
+        // Refresh the clock if the seconds have changed.
+        const time_t displayed_time = now.tv_sec;
         gettimeofday(&now, NULL);
-
-        // Refresh the clock on timeouts.
-        if (select_ret == 0)
+        if (now.tv_sec != displayed_time)
```

Comment:
I wonder if re-adding `|| (select_ret == 0)` here would, hurt or help or even be necessary. Guts say "help". What do you think?

Reply:
If `now.tv_sec != displayed_time` is `false`, then we know for sure that the clock does not need to be re-rendered, so adding `|| (select_ret == 0)` would not help.